Lib/tarfile.py

```diff
@@ -1468,7 +1468,7 @@ def __init__(self, name=None, mode="r", fileobj=None, format=None,
             if hasattr(fileobj, "mode"):
                 self._mode = fileobj.mode
             self._extfileobj = True
-        self.name = os.path.abspath(name) if name else None
+        self.name = os.fspath(name) if name else None
```

Comment:
these lines are not equal.
The correct code should be something like:
```
name = os.fspath(name)
self.name = os.path.abspath(name)
```
I've skipped the check for empty/None name.